### DISCUSSION TITLE
Uploading Large Folders

### DISCUSSION THREAD

**Discussion by Crossbow013** [community/community#40872] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

**THANK YOU!!** - for taking the time to read and try to answer my question.  I'm trying to upload a project I have but have been unable to figure out how to upload folders.  I searched the discussions and found an article that mentioned just dropping the files in the repo.  I tried dragging and dropping a project folder into my repo, but it said the file was too large and to try to limit it to 100 files.  I am not sure (don't want) to upload files individually given the number there are.

The project is a game I made in RPG Maker MZ.  I'm also trying to figure out from that community if I can limit the files (such as data) to lessen the amount to upload.  I'm new to GitHub, so my apologies if I'm asking something that is inappropriate or answered somewhere else.  Thanks again and I hope you have a wonderful day!

**Comment by GabyCT** [comment, marked as answer]
There are two ways to add a file to a repository https://docs.github.com/en/repositories/working-with-files/managing-files/adding-a-file-to-a-repository, the first one is to add and drop on the browser however it is limited to 100 MB for larger files you can see https://docs.github.com/en/repositories/working-with-files/managing-large-files/about-large-files-on-github. And the other way is do that via command line: 
```
On your computer, move the file you'd like to upload to GitHub into the local directory that was created when you cloned the repository.
1. Open Git Bash.
2. Change the current working directory to your local repository.
3. Stage the file for commit to your local repository.
$ git add .
# Adds the file to your local repository and stages it for commit. To unstage a file, use 'git reset HEAD YOUR-FILE'.
4. Commit the file that you've staged in your local repository.
$ git commit -m "Add existing file"
# Commits the tracked changes and prepares them to be pushed to a remote repository. To remove this commit and modify the file, use 'git reset --soft HEAD~1' and commit and add the file again.
[Push the changes](https://docs.github.com/en/github/getting-started-with-github/pushing-commits-to-a-remote-repository) in your local repository to GitHub.com.
$ git push origin YOUR_BRANCH
# Pushes the changes in your local repository up to the remote repository you specified as the origin
```

**Reply by Crossbow013** [reply]
Thanks for such a quick and thorough reply.  I will check out the links and give them a try.

**Reply by Crossbow013** [reply]
Hi, I read the article and am now trying to work through your instructions... 

I got stuck on #3... How do I stage files that I haven't changed?  I know how to add and commit files I've made changes to, but these files/folders are from several months ago.  There are around 30 files/folders in the main folder I am trying to upload.  I'm familiar with step 4, so I'm really only stuck on 3.

Thanks again!

**Reply by GabyCT** [reply]
To add files that are new or changed you still need to to `git add` for example...I have 30 files under a directory called `test` I will do something like this
```
$ git add /home/ubuntu/test/* 
```

**Comment by batta77** [comment]
> ### Select Topic Area
> Question
> 
> ### Body
> **THANK YOU!!** - for taking the time to read and try to answer my question. I'm trying to upload a project I have but have been unable to figure out how to upload folders. I searched the discussions and found an article that mentioned just dropping the files in the repo. I tried dragging and dropping a project folder into my repo, but it said the file was too large and to try to limit it to 100 files. I am not sure (don't want) to upload files individually given the number there are.
> 
> The project is a game I made in RPG Maker MZ. I'm also trying to figure out from that community if I can limit the files (such as data) to lessen the amount to upload. I'm new to GitHub, so my apologies if I'm asking something that is inappropriate or answered somewhere else. Thanks again and I hope you have a wonderful day!

**Reply by Crossbow013** [reply]
Hi batta77,

Not sure if you meant to or not, but I am only able to see my post in your response... did you mean to do that, or was there more to the message that didn't come through?

**Comment by Crossbow013** [comment]
Thanks for all the great information!!!

**Comment by mrb-sage** [comment]
My Project is a unreal game project and i cant upload the whole project, its about 50GB that i need to upload ! HELP!
